### PR TITLE
Add an Android HttpEngine regression test

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
@swankjesse requesting post review.

Just a test we can compare any reported bugs against Android HttpEngine

File: regression-test/src/androidTest/java/okhttp/regression/compare/AndroidHttpEngineTest.kt [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+          override fun onReadCompleted(
+            request: UrlRequest,
+            info: UrlResponseInfo,
+            byteBuffer: ByteBuffer,
+          ) {
+            println("onReadCompleted ${info.headers.asMap}")
+            byteBuffer.flip()
+            buffer.write(byteBuffer)
+            byteBuffer.clear()
+            request.read(byteBuffer)
```
Comment (swankjesse) [inline regression-test/src/androidTest/java/okhttp/regression/compare/AndroidHttpEngineTest.kt:148]: Off topic: I find this API to be completely confusing

**Review by swankjesse — COMMENTED** [review]
LGTM

File: regression-test/src/androidTest/java/okhttp/regression/compare/AndroidHttpEngineTest.kt (hunk 1 above)
Comment (yschimke) [reply regression-test/src/androidTest/java/okhttp/regression/compare/AndroidHttpEngineTest.kt:148]: Yeah, agreed. I struggled to write this and I've used it before.

It is possibly optimal with zero copy transfer etc. and you can either consume like I have here as response bytes are received, or out of line as a caller requests them.

**Pull request merged**